src/zkscript/groth16/model/groth16.py

```diff
@@ -156,21 +176,26 @@ def groth16_verifier_unlock(
         max_multipliers: list[int] | None = None,
         load_q=True,
     ) -> Script:
-        r"""Generate unlocking script for groth16_verifier.
-
-        - pub: list of public statements
-        - (A,B,C): zk-proof, elements passed as their list of coordinates
-        - lambdas_B_exp_miller_loop: gradients needed to compute val * B, see also unrolled_multiplication in
-        EllipticCurveFqUnrolled (val is the value over which we compute the Miller loop)
-        - lambdas_minus_gamma_exp_miller_loop: gradients needed to compute val * (-gamma)
-        - lambdas_minus_delta_exp_miller_loop: gradients needed to compute val * (-delta)
-        - inverse_miller_loop: inverse of miller_loop(A,B) * miller_loop(C,-gamma) * miller_loop(sum_gamma_abc,-delta),
-        where gamma_abc is taken from the vk
-        - lamdbas_partial_sums: list of gradients, the element at position n_pub - i - 1 is the list of gradients to
-        compute a_(i+1) * gamma_abc[i] and \sum_(j=0)^(i) a_j * gamma_abc[j], 0 <= i <= n_pub - 1
-        - lambdas_multiplications: list of gradients, the element at position i is the list of gradients to compute
-        pub[i] * gamma_abc[i], 0 <= i <= n_pub - 1
-        - max_multipliers[i]: upper bound for public statement pub[i]
+        r"""Generate the unlocking script for the Groth16 verifier.
+
+        Args:
+            pub (list[int]): List of public statements.
+            A (list[int]): Point A of the zk-proof. Specified as a list of coordinates.
+            B (list[int]): Point B of the zk-proof. Specified as a list of coordinates.
+            C (list[int]): Point C of the zk-proof. Specified as a list of coordinates.
+            lambdas_B_exp_miller_loop (list[list[list[int]]]): Gradients needed to compute val * B, see also
+                unrolled_multiplication in EllipticCurveFqUnrolled (val is the value over which we compute the Miller
+                loop).
+            lambdas_minus_gamma_exp_miller_loop (list[list[list[int]]]): Gradients needed to compute val * (-gamma).
+            lambdas_minus_delta_exp_miller_loop (list[list[list[int]]]): Gradients needed to compute val * (-delta).
+            inverse_miller_loop (list[int]): Inverse of miller_loop(A,B) * miller_loop(C,-gamma) *
+                miller_loop(sum_gamma_abc,-delta), where gamma_abc is taken from the vk.
```

Comment:
Shouldn't this be inverse of miller_loop(A,B) * miller_loop(C,-delta) * miller_loop(sum_gamma_abc,-gamma) @BarbacoviF ?

Reply:
Agreed, good catch!